lib/SILOptimizer/IPO/GlobalOpt.cpp

```diff
@@ -1455,10 +1455,13 @@ bool SILGlobalOpt::run() {
     GlobIdx = 0;
     for (auto &BB : F) {
       bool IsCold = ColdBlocks.isCold(&BB);
-      auto Iter = BB.begin();
-      while (Iter != BB.end()) {
-        SILInstruction *I = &*Iter;
-        Iter++;
+
+      llvm::SmallVector<SILInstruction *, 4> Worklist;
+      for (SILInstruction &SI : reverse(BB))
```

Comment:
I used reverse here just  to be consistent. I think we could scan the BB in the opposite direction and the dataflow problem would be actually equivalent. Haven't given it much thought and wanted to limit the functional changes to the fix. I don't think it matters in practice, FWIW.